scipy/interpolate/_interpolate.py

```diff
@@ -499,7 +499,7 @@ def __init__(self, x, y, kind='linear', axis=-1,
         _Interpolator1D.__init__(self, x, y, axis=axis)
 
         self.bounds_error = bounds_error  # used by fill_value setter
-        self.copy = copy
+        self.copy = np._CopyMode.ALWAYS if copy else np._CopyMode.IF_NEEDED
```

Comment:
This could use a comment like:
```python
# `copy` keyword semantics changed in NumPy 2.0, once that is the minimum version
# this can use `copy=None`.
```

Reply:
Done!